core/src/main/scala/org/bykn/bosatsu/Normalization.scala

```diff
@@ -0,0 +1,241 @@
+package org.bykn.bosatsu
+
+import cats.data.{NonEmptyList, State}
+import com.stripe.dagon.Memoize
+import cats.Id
+import cats.implicits._
+
+sealed abstract class NormalExpression {
+
+}
+
+object NormalExpression {
```

Comment:
this seems like it could be a separate PR that we could test independently, which would be nice.